New Template support for broadcom_icos as a new OS and added show_version command

alepodj opened the pull request:
Added support for broadcom_icos as a new OS and added show_version command

##### ISSUE TYPE
<!--- Pick one below and delete the rest: -->
 - New Template Pull Request

##### COMPONENT
<!--- Name of the template, os and command  -->

```
template --> broadcom_icos_show_version.textfsm
os --> broadcom_icos
cmd --> show version
```

##### SUMMARY
<!--- Describe the change, including rationale and design decisions -->

As discussed in https://github.com/networktocode/ntc-templates/issues/726 support for Broadcom Icos devices is not present current in ntc_templates. This is the first attemp at adding the os all together as well as the 1st command: show version.

Even tho the following models share the same OS(icos), is expected, much like anywhere else, that each hardware model may display different values based on the model the command is ran on. For that i included 5 different outputs from 5 different hardware models in order to be as inclusive with the captures as possible. The models available to me are below and the all run ICOS in a version or another, as described in the show version command itself:

```
1- Accton AS4610-54P
2- Data Center Switch Software AS5610-52X(technically Accton AS5610-52X)
3- Quanta LY2R
4- Quanta LB9
5- DNI-3448P

Potentially i may also be able to get a Quanta LB8 to be even more inclusive
```

FragmentedPacket commented on templates/broadcom_icos_show_version.textfsm:
```diff
@@ -0,0 +1,41 @@
+  ^\s*Board\s*Revision\s*\.+\s*${Board_Revision}$$
+  ^\S \S\S -> Continue.Record
+  ^\s*Additional\s*Packages\s*\.+\s*${Additional_Packages}$$
+  ^\s*${Additional_Packages}$$
```
Just to be safe, I would do this:
```suggestion
  ^\s+${Additional_Packages}$$
```

alepodj replied:
Replaced as suggested

FragmentedPacket commented on templates/broadcom_icos_show_version.textfsm:
```diff
@@ -0,0 +1,41 @@
+Value System_Description (.+)
```
Can you please make all the values upper-case?

```
Value SYSTEM_DESCRIPTION (.+)
```

alepodj replied:
Corrected all named values to upper-case

FragmentedPacket commented on templates/broadcom_icos_show_version.textfsm:
```diff
@@ -0,0 +1,41 @@
+Value System_Description (.+)
+Value Machine_Type (.+)
+Value Machine_Model (.+)
+Value Serial_Number (.+)
```
Could you also look at other `show_version` from other platforms and make sure you're using the same named values as those? This helps bring our templates more vendor agnostic in the way they return the data. Users can act on the same keys for other platforms.

alepodj replied:
Went thru several show versions and took a hint on regularly used named values. A recent Ubiquity pull showed very similar patterns so used as a guide, mainly: https://github.com/networktocode/ntc-templates/tree/master/tests/ubiquiti_edgeswitch/show_version but also other common names were replaced. No references were found for the rest so, left as is or shortened if the name made sense still.

FragmentedPacket commented on templates/broadcom_icos_show_version.textfsm:
```diff
@@ -0,0 +1,41 @@
+  ^\s*Network\s*Processing\s*Device\s*\.+\s*${Network_Processing_Device}$$
+  ^\s*CPLD\s*version\s*\.+\s*${CPLD_version}$$
+  ^\s*Board\s*Revision\s*\.+\s*${Board_Revision}$$
+  ^\S \S\S -> Continue.Record
```
```
  ^\S \S\S -> Continue.Record
```
What does this line account for?

alepodj replied:
A misunderstanding and its now removed

alepodj commented on templates/broadcom_icos_show_version.textfsm:
```diff
@@ -0,0 +1,41 @@
+  # Accton AS4610-54P, Accton AS5610-52X, Quanta LY2R, Quanta LB9, DNI AG3448P-R   
+  # The following can be an empty value as it doesnt exist in all the models:
+  # FruNumber, PartNumber, CPLDversion, BoardRevision
+  ^\s*$$
```
Removed 1st:  ^\s*$$

alepodj commented on templates/broadcom_icos_show_version.textfsm:
```diff
@@ -0,0 +1,41 @@
+  ^\s*Board\s*Revision\s*\.+\s*${Board_Revision}$$
+  ^\S \S\S -> Continue.Record
+  ^\s*Additional\s*Packages\s*\.+\s*${Additional_Packages}$$
+  ^\s*${Additional_Packages}$$
```
Added ^. -> Error

alepodj commented:
Corrected and modified as requested and committed the new changes https://github.com/networktocode/ntc-templates/pull/733/commits/4832da5fa840d73aee04deb9c377f05bc25182f0
